sqlglot/dialects/bigquery.py

```diff
@@ -997,20 +1027,38 @@ def _parse_features_at_time(self) -> exp.FeaturesAtTime:
 
             return expr
 
-        def _parse_generate_embedding(self) -> exp.GenerateEmbedding:
-            self._match_text_seq("MODEL")
-            this = self._parse_table()
+        def _parse_vector_search(self) -> exp.VectorSearch:
+            base_table = (
+                self._match(TokenType.TABLE) and self._parse_table()
+            ) or self._parse_select(nested=True)
```

Comment:
Let's change this a little to follow the existing pattern:

```python
self._match(TokenType.TABLE)
base_table = self._parse_table()  # Can consume a derived table (query) too
...
```

Reply:
@VaggelisD do you remember why we did `(...) or _parse_select`? Is there something not handled by `_parse_table` that you had discovered?